src/core/scope.ml

```diff
@@ -73,11 +73,16 @@ type mode =
   (** Scoping mode for rewriting rule left-hand sides. The constructor carries
       a flag that is set to [true] if {!constructor:Terms.expo.Privat} symbols
       are allowed, and also additional data. *)
-  | M_RHS  of bool * (string, tevar) Hashtbl.t
-  (** Scoping mode for rewriting rule righ-hand sides. The constructor carries
-      a flag that is set to [true] if {!constructor:Terms.expo.Privat} symbols
-      are allowed, and the environment for variables that we known to be bound
-      in the RHS. *)
+  | M_RHS  of
+      { m_r_p             : bool
+      (** True if {!constructor:Terms.expo.Privat} symbols are allowed. *)
+      ; m_r_data          : (string, tevar) Hashtbl.t
+      (** Environment for variables that we know to be bound in the RHS. *)
+      ; mutable m_r_xvars : (int * tevar list)
```

Comment:
Why not splitting in two fields? This would avoid to have to do match's and fst/snd's.

Reply:
I wanted to pack the mutable elements, as you wish.

Reply:
I'm using two fields now.